components/homme/src/share/cxx/Dimensions.hpp

```diff
@@ -14,7 +14,7 @@
 namespace Homme {
 
 // Until whenever CUDA supports constexpr properly
-#ifdef CUDA_BUILD
+#ifdef HOMMEXX_ENABLE_GPU
```

Comment:
I admit i don't know what changes you propose.

Reply:
I am wondering if the code would compile on hip/cuda if we removed the code inside `#ifdef CUDA_BUILD` and leave only the code in the `#else` branch. Back in the hommexx dev days, we added this ifdef because Cuda did not like the `static constexpr` variables (I don't recall what the error was, likely some undefined references). But I wonder if with c++17 it would now work.